apex/contrib/csrc/groupbn/batch_norm_add_relu.cu

```diff
@@ -85,21 +85,23 @@ at::Tensor nhwc_bn_addrelu_fwd_train(
   // Create wrapper
   NhwcBatchNormAddRelu *bn = new NhwcBatchNormAddRelu();
 
-  bn->setInputDescriptor(CUDNN_TENSOR_NHWC, CUDNN_DATA_HALF, N, C, H, W, bn_group);
-  bn->setOutputDescriptor(CUDNN_TENSOR_NHWC, CUDNN_DATA_HALF, N, C, H, W);
+  bn->setInputDescriptor(DNN_TENSOR_FORMAT, DNN_DATA_HALF, N, C, H, W, bn_group);
+  bn->setOutputDescriptor(DNN_TENSOR_FORMAT, DNN_DATA_HALF, N, C, H, W);
 
   bn->setConstants(momentum, epsilon);
 
   // set pointers within the wrapper
-  bn->setInputOutputPointers(x.DATA_PTR<at::Half>(),
+  bn->setInputOutputPointers(x.contiguous().DATA_PTR<at::Half>(),
                              nullptr,
                              y.DATA_PTR<at::Half>(),
```

Comment:
@sarunyap Could you please explain the reasoning behind using `.contiguous()` on all the tensors here? Does `y` not need to be contiguous?

Reply:
Tensor x is the "input" tensor that is given by a user which can be non-contiguous.  Tensor y is created inside the function which is always contiguous.  So, tensor y does not need to be made contiguous before being used.